src/code/z_lifemeter.c

```diff
@@ -1,17 +1,446 @@
+            if ((ddCount < 0) || (i > ddCount)) {
+                if (curCombineModeSet != 1) {
+                    curCombineModeSet = 1;
+                    func_8012C654(gfxCtx);
+                    gDPSetCombineLERP(OVERLAY_DISP++, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE,
+                                      0, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0);
+                }
+            } else {
+                if (curCombineModeSet != 3) {
+                    curCombineModeSet = 3;
+                    func_8012C654(gfxCtx);
+                    gDPSetCombineLERP(OVERLAY_DISP++, ENVIRONMENT, PRIMITIVE, TEXEL0, PRIMITIVE, TEXEL0, 0, PRIMITIVE,
+                                      0, ENVIRONMENT, PRIMITIVE, TEXEL0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0);
+                }
+            }
+            {
+                posY = 26.0f + offsetY;
+                posX = 30.0f + offsetX;
+                temp_f4 = 1.0f;
+                temp_f4 /= 0.68f;
+                temp_f4 *= 1 << 10;
+                halfTexSize = 8.0f;
+                halfTexSize *= 0.68f;
+                gSPTextureRectangle(OVERLAY_DISP++, (s32)((posX - halfTexSize) * 4), (s32)((posY - halfTexSize) * 4),
+                                    (s32)((posX + halfTexSize) * 4), (s32)((posY + halfTexSize) * 4), G_TX_RENDERTILE,
+                                    0, 0, (s32)temp_f4, (s32)temp_f4);
+            }
+        } else {
+            if ((ddCount < 0) || (ddCount < i)) {
+                if (curCombineModeSet != 2) {
+                    curCombineModeSet = 2;
+                    func_8012C8D4(gfxCtx);
+                    gDPSetCombineLERP(OVERLAY_DISP++, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE,
+                                      0, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0);
+                    gDPSetAlphaCompare(OVERLAY_DISP++, G_AC_THRESHOLD);
+                }
+            } else {
+                if (curCombineModeSet != 4) {
+                    curCombineModeSet = 4;
+                    func_8012C8D4(gfxCtx);
+                    gDPSetCombineLERP(OVERLAY_DISP++, ENVIRONMENT, PRIMITIVE, TEXEL0, PRIMITIVE, TEXEL0, 0, PRIMITIVE,
+                                      0, ENVIRONMENT, PRIMITIVE, TEXEL0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0);
+                    gDPSetAlphaCompare(OVERLAY_DISP++, G_AC_THRESHOLD);
+                }
+            }
+            {
+                Mtx* temp_a0 = GRAPH_ALLOC(gfxCtx, sizeof(Mtx));
+                func_801780F0(temp_a0, 1.0f - (0.32f * lifesize), 1.0f - (0.32f * lifesize), 1.0f - (0.32f * lifesize),
+                              -130.0f + offsetX, 94.5f - offsetY, 0.0f);
+                gSPMatrix(OVERLAY_DISP++, temp_a0, G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPVertex(OVERLAY_DISP++, beatingHeartVtx, 4, 0);
+                gSP1Quadrangle(OVERLAY_DISP++, 0, 2, 3, 1, 0);
+            }
```

Comment:
```suggestion
            {
                Mtx* mtx = GRAPH_ALLOC(gfxCtx, sizeof(Mtx));
                
                func_801780F0(mtx, 1.0f - (0.32f * lifesize), 1.0f - (0.32f * lifesize), 1.0f - (0.32f * lifesize),
                              -130.0f + offsetX, 94.5f - offsetY, 0.0f);
                gSPMatrix(OVERLAY_DISP++, mtx, G_MTX_LOAD | G_MTX_MODELVIEW);
                gSPVertex(OVERLAY_DISP++, beatingHeartVtx, 4, 0);
                gSP1Quadrangle(OVERLAY_DISP++, 0, 2, 3, 1, 0);
            }
```
Oof, this extra scope doesn't look pretty but I'm guessing it's needed to match? There's definitely been much worse though in draw functions

Reply:
it doesn't match if mtx is declared at the start of the function ( because it changes the stackpointer), but it can also match declaring it at the start of the else block : 
```C
else {
    Mtx* mtx;
    /*
     non relevant part
    */
    mtx = GRAPH_ALLOC(gfxCtx, sizeof(Mtx));

    func_801780F0(mtx, 1.0f - (0.32f * lifesize), 1.0f - (0.32f * lifesize), 1.0f - (0.32f * lifesize),
                  -130.0f + offsetX, 94.5f - offsetY, 0.0f);
    gSPMatrix(OVERLAY_DISP++, mtx, G_MTX_LOAD | G_MTX_MODELVIEW);
    gSPVertex(OVERLAY_DISP++, beatingHeartVtx, 4, 0);
    gSP1Quadrangle(OVERLAY_DISP++, 0, 2, 3, 1, 0);            
}
```
Not sure which one is prefered here